src/sign.rs

```diff
@@ -773,6 +801,36 @@ mod tests {
         );
     }
 
+    #[test]
+    fn cose_sign1_ec256_text_with_extra_protected() {
+        let (ec_private, ec_public) = generate_ec256_test_key();
+
+        let mut protected = HeaderMap::new();
+        protected.insert(CborValue::Integer(1), CborValue::Bytes(b"11".to_vec()));
+        protected.insert(CborValue::Integer(15), CborValue::Bytes(b"12".to_vec()));
+
+        let mut unprotected = HeaderMap::new();
+        unprotected.insert(CborValue::Integer(4), CborValue::Bytes(b"11".to_vec()));
+
+        let cose_doc1 =
+            COSESign1::new_with_protected(TEXT, &protected, &unprotected, &ec_private).unwrap();
+        let cose_doc2 = COSESign1::from_bytes(&cose_doc1.as_bytes(false).unwrap()).unwrap();
+
+        let (protected, payload) = cose_doc2
+            .get_protected_and_payload(Some(&ec_public))
+            .unwrap();
+
+        assert_eq!(
+            protected.get(&CborValue::Integer(1)),
+            Some(&CborValue::Integer(-7)),
+        );
```

Comment:
This behavior is slightly unexpected.

I think `new_with_protected` should either:
* copy the protected header as is and not add or remove anything from it
* add additional fields into protected, but only if there is no conflict; fail if there is a conflict
* copy the protected header as is, but verify the variables it sets and fail if it thinks there's an error

I like the first option better, and `new` should just set the protected fields as it did until now.

Reply:
Thanks for the comments. I implemented the first option.